### ISSUE TITLE
Understanding Deep Learning Requires Rethinking Gerneralization

### ISSUE THREAD

**Issue by reyllama** [reyllama/papers#4]
## Authors <br>
Chiyuan Zhang, Samy Bengio, Moritz Hardt, Benjamin Recht, Oriol Vinyals

## Tags <br>
- Generalization <br>
- Regularization <br>
- Convolutional Neural Net <br>
- Deep Learning Architecture

## Summary <br>
- Incisive reflection on **Generalization Performance** of modern deep learning models.
- As state-of-the-art deep neural networks possess far greater number of parameters than the size of dataset, **Effective Capacity** of these models exceed the dataset to the point where it can literally "memorize" them all.
> - Even when there is **NO correlation** between image and its label due to intentional randomization of labels/input, - thereby making it **impossible** to learn anything at all, in human sense - the model can still fit the training data. 
> - We can attribute the satisfying generalization performance of successful NNs no more to the fundamental concept of **learning representations** itself.
- Regularizers, both explicit (e.g. Data Augmentation, Weight Decay, Dropout) and implicit (e.g. Batch Normalization, Early Stopping) contribute to generalization performance **to certain extent in certain situation**. However, they don't look like the primary drive on the issue, since models continue to generalize well even without them.
- [quote] "There exists a two-layer neural network with ReLU activations and 2n+d weights that
can represent any function on a sample of size n in d dimensions."
- It turns out to be the case that we cannot explain the generalization capacity of linear models well enough as well. Linear models, with the help of Stochastic Gradient Descent, can perform well on test cases even without regularizations. (Linear Algebra)

![image](https://user-images.githubusercontent.com/46308407/77498609-63183480-6e93-11ea-9832-5bea6237c132.png)

> - Models can fit perfectly well for any training samples with or without true labels/regularizations.
> - While regularization does help generalization performance overall, the effect is not dramatic.

## Personal Comments <br>
- Why is it possible that deep neural networks learn something? How can the findings be generalized? We still have a long way to clear understanding.
- I still do think that the accuracy score boost from regularization is rather significant, it was interesting that with SGD even as simple a model as linear model can generalize well without the help of explicit regularization.
- I must grow patience for understanding Math in papers. There's nothing too difficult to understand, but I seem to lose the grip whenever math formulas come up.
- It was quite interesting overall for that it tackled the issue of DL interpretability/explainability.
